src/nvim/window.c

```diff
@@ -1010,6 +1010,10 @@ void ui_ext_win_position(win_T *wp, bool validate)
       comp_col += grid->comp_col;
       comp_row = MAX(MIN(comp_row, Rows - wp->w_height_outer - (p_ch > 0 ? 1 : 0)), 0);
       comp_col = MAX(MIN(comp_col, Columns - wp->w_width_outer), 0);
+      int right_extra = Columns - (int)c.col - wp->w_width - (c.border_chars[2][0] != 0);
+      if (!(c.anchor & kFloatAnchorEast) && c.fixed && !wp->w_p_wrap && right_extra < 0) {
```

Comment:
Why check for `w_p_wrap`? This sounds like something that should be checked by the called that sets `fixed`.

Reply:
cause vim help said there check wrap . when wrap on it will not fixed float window width on screen.

```
                         - "pos" is "botleft" or "topleft", and
                         - "wrap" is off, and
                         - the popup would be truncated at the right edge of
                           the screen, then 
```

Reply:
This sounds like a completely unnecessary restriction. I don't think it makes sense.

Reply:
removed.